lib/main.js

```diff
@@ -107,52 +131,84 @@ HTMLUglify.prototype.setNewSelector = function(lead, rule, toReplace, pointer) {
+          value = selectorPart.replace(/\#/g, '');
+
+          if (value && !self.isWhitelisted('id', value)) {
+            pointer = self.determinePointer('id', value, lookups);
+            self.populateLookups('id', value, pointer, lookups);
+          }
         }
 
         // handle .something
         if (selectorPart.indexOf('.') > -1) {
-          toReplace = selectorPart.replace(/\./g, '');
-          keyLookup = ['class', toReplace].join(LOOKUP_DELIMITER);
+          value = selectorPart.replace(/\./g, '');
+
+          if (value && !self.isWhitelisted('class', value)) {
+            pointer = self.determinePointer('class', value, lookups);
+            self.populateLookups('class', value, pointer, lookups);
+          }
         }
 
-        // handle *[id=something]
-        if (selectorPart.indexOf('id=') > -1){
-          toReplace = selectorPart.replace(/id=/g, '').replace(/\"/g, ''); // remove quotes from "somevalue"
-          keyLookup = ['id', toReplace].join(LOOKUP_DELIMITER);
+        // handle *[id=something] or *[id*=something]
+        if (
+          selectorPart.indexOf('id=') > -1 || 
+          selectorPart.indexOf('id*=') > -1 ||
+          selectorPart.indexOf('id^=') > -1 ||
+          selectorPart.indexOf('id$=') > -1
+        ) {
+          value = selectorPart.replace(/id\*?\^?\$?=|\"|\'/g, '');
+          if (value && !self.isWhitelisted('id', value)) {
+            pointer = self.determinePointer('id', value, lookups);
+            self.populateLookups('id', value, pointer, lookups);
+          }
         }
 
-        // handle *[class=something]
-        if (selectorPart.indexOf('class=') > -1) {
-          toReplace = selectorPart.replace(/class=/g, '');
-          keyLookup = ['class', toReplace].join(LOOKUP_DELIMITER);
+        // handle *[class=something] or *[class*=something]
+        if (
+          selectorPart.indexOf('class=') > -1 || 
+          selectorPart.indexOf('class*=') > -1 ||
+          selectorPart.indexOf('class^=') > -1 ||
+          selectorPart.indexOf('class$=') > -1
+        ) {
+          value = selectorPart.replace(/class\*?\^?\$?=|\"|\'/g, '');
+
+          if (value && !self.isWhitelisted('class', value)) {
+            pointer = self.determinePointer('class', value, lookups);
+            self.populateLookups('class', value, pointer, lookups);
+          }
         }
 
-        // handle *[for=something]
-        if (selectorPart.indexOf('for=') > -1){
-          toReplace = selectorPart.replace(/for=/g, '');
-          toReplace = toReplace.replace(/\"/g, ''); // remove quotes from "somevalue"
-          keyLookup = ['id', toReplace].join(LOOKUP_DELIMITER);
+        // handle *[for=something] or *[for*=something]
+        if (
+          selectorPart.indexOf('for=') > -1 || 
+          selectorPart.indexOf('for*=') > -1 ||
+          selectorPart.indexOf('for^=') > -1 ||
+          selectorPart.indexOf('for$=') > -1
```

Comment:
I guess we can revisit once we're using a real CSS selector parser (e.g. https://github.com/postcss/postcss-selector-parser) because it's starting to get insane.